gcloud-java-compute/src/main/java/com/google/gcloud/compute/Snapshot.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Google Compute Engine snapshot. Compute Engine allows you to take snapshots of your persistent
+ * disk and create new persistent disks from that snapshot. This can be useful for backing up data,
+ * recreating a persistent disk that might have been lost, or copying a persistent disk. Snapshots
+ * can be applied across persistent disk types. {@code Snapshot} adds a layer of service-related
+ * functionality over {@link SnapshotInfo}. Obejcts of this class are immutable, to get a
+ * {@code Snapshot} object with the most recent information use {@link #reload}.
+ *
+ * @see <a href="https://cloud.google.com/compute/docs/disks/persistent-disks#snapshots">Use
+ *     persistent disk snapshots</a>
+ */
+public class Snapshot extends SnapshotInfo {
+
+  private static final long serialVersionUID = -973924811396336695L;
+
+  private final ComputeOptions options;
+  private transient Compute compute;
+
+  /**
+   * A builder for {@code Snapshot} objects.
+   */
+  public static class Builder extends SnapshotInfo.Builder {
```

Comment:
Yes the `toBuilder()` method is there mainly to be able to write something like:
```java
Snapshot snapshot = compute.get("some-snapshot");
if (snapshot.creationTimestamp() < someDate) {
  compute.create(snapshot.toBuilder().snapshotId(SnapshotId.of("some-recent-snapshot")).build());
}
```
This is no killer feature but it comes at no cost. Given that we have `toBuilder()` in `SnapshotInfo` not redefining it for `Snapshot` would be counter-intuitive. We could consider removing both if you believe this is of no use.